core/EE_System.core.php

```diff
@@ -684,82 +620,76 @@ public function is_major_version_change()
-     * @param string $version_to_upgrade_to            the version that was just upgraded to (for core that will be
+     * @param string $current_version                  the version that was just upgraded to (for core that will be
      *                                                 espresso_version())
-     * @return int one of the constants on EE_System::req_type_*
+     * @return int one of the constants on EE_System::req_type_
      */
     public static function detect_req_type_given_activation_history(
-        $activation_history_for_addon,
-        $activation_indicator_option_name,
-        $version_to_upgrade_to
-    ) {
-        $version_is_higher = self::_new_version_is_higher($activation_history_for_addon, $version_to_upgrade_to);
-        if ($activation_history_for_addon) {
-            // it exists, so this isn't a completely new install
-            // check if this version already in that list of previously installed versions
-            if (! isset($activation_history_for_addon[ $version_to_upgrade_to ])) {
-                // it a version we haven't seen before
-                if ($version_is_higher === 1) {
-                    $req_type = EE_System::req_type_upgrade;
-                } else {
-                    $req_type = EE_System::req_type_downgrade;
-                }
-                delete_option($activation_indicator_option_name);
-            } else {
-                // its not an update. maybe a reactivation?
-                if (get_option($activation_indicator_option_name, false)) {
-                    if ($version_is_higher === -1) {
-                        $req_type = EE_System::req_type_downgrade;
-                    } elseif ($version_is_higher === 0) {
-                        // we've seen this version before, but it's an activation. must be a reactivation
-                        $req_type = EE_System::req_type_reactivation;
-                    } else {// $version_is_higher === 1
-                        $req_type = EE_System::req_type_upgrade;
-                    }
-                    delete_option($activation_indicator_option_name);
-                } else {
-                    // we've seen this version before and the activation indicate doesn't show it was just activated
-                    if ($version_is_higher === -1) {
-                        $req_type = EE_System::req_type_downgrade;
-                    } elseif ($version_is_higher === 0) {
-                        // we've seen this version before and it's not an activation. its normal request
-                        $req_type = EE_System::req_type_normal;
-                    } else {// $version_is_higher === 1
-                        $req_type = EE_System::req_type_upgrade;
-                    }
-                }
-            }
-        } else {
-            // brand new install
-            $req_type = EE_System::req_type_new_activation;
```

Comment:
after much analyzation... I realized this could all be replaced by the 12 lines of code in the new `getRequestType()` function i added below